signer/src/main/kotlin/io/provenance/hdwallet/signer/SignatureFormat.kt

```diff
@@ -0,0 +1,214 @@
+package io.provenance.hdwallet.signer
+
+import io.provenance.hdwallet.ec.Curve
+import java.io.ByteArrayOutputStream
+import java.math.BigInteger
+import org.bouncycastle.asn1.ASN1InputStream
+import java.security.Signature as JavaSignature
+import org.bouncycastle.asn1.ASN1Integer
+import org.bouncycastle.asn1.ASN1Primitive
+import org.bouncycastle.asn1.ASN1Sequence
+import org.bouncycastle.asn1.DERSequenceGenerator
+
+@JvmInline
+value class BTCSignature private constructor(private val value: ByteArray) {
+    companion object {
+        fun fromByteArray(bytes: ByteArray): BTCSignature = BTCSignature(bytes)
+    }
+
+    fun toByteArray(): ByteArray = value
+}
+
+@JvmInline
+value class ASN1Signature private constructor(private val value: ByteArray) {
+    companion object {
+        fun fromByteArray(bytes: ByteArray): ASN1Signature = ASN1Signature(bytes)
+    }
+
+    fun toByteArray(): ByteArray = value
+    fun toIntegerPair(): BigIntegerPair = ASN1.decode(this)
+}
+
+@JvmInline
+value class BigIntegerPair(private val intPair: Pair<BigInteger, BigInteger>) {
+    val r get(): BigInteger = intPair.first
+    val s get(): BigInteger = intPair.second
+    fun asPair(): Pair<BigInteger, BigInteger> = intPair
+}
+
+/**
+ * Bitcoin-specific signature format operations.
+ */
+object BTC {
+    /**
+     * [encode] returns the ECDSA signature as a ByteArray of `r || s`, * where both `r` and `s` are encoded into 32
+     * byte big endian integers.
+     */
+    fun encode(r: BigInteger, s: BigInteger, curve: Curve): BTCSignature {
```

Comment:
I have a small nitpick, does it make sense to call this a bitcoin signature? Isn't the R and S pair just a specific encoding of an ECDSA signature and not necessarily bitcoin specific? Not sure if there could be a better naming convention

Reply:
Yea, they are technically called `canonical` ecdsa signatures (it's mentioned all over in go's [ecdsa](https://pkg.go.dev/github.com/decred/dcrd/dcrec/secp256k1/v4/ecdsa#section-readme))